packages/iov-ledger-bns/src/app.ts

```diff
@@ -13,16 +13,19 @@ const cmdPubkeyWithPath = 5;
 const cmdAppVersion = 0xca;
 
 export function getPublicKey(transport: TransportNodeHid): Promise<Uint8Array> {
-  return sendChunks(transport, appCode, cmdPubkey, new Uint8Array([]));
+  return sendChunks(transport, appCode, cmdPubkey, new Uint8Array([0]));
 }
 
 export function getPublicKeyWithIndex(transport: TransportNodeHid, i: number): Promise<Uint8Array> {
   const pathComponent = Slip10RawIndex.hardened(i).asNumber();
-  return sendChunks(transport, appCode, cmdPubkeyWithPath, encodeUint32(pathComponent));
+  let pathPart = encodeUint32(pathComponent);
+  pathPart = [pathComponent.length, ...pathPart];
+  return sendChunks(transport, appCode, cmdPubkeyWithPath, pathPart);
 }
 
 export function signTransaction(transport: TransportNodeHid, transaction: Uint8Array): Promise<Uint8Array> {
-  return sendChunks(transport, appCode, cmdSign, transaction);
+  const paddedTx: ReadOnlyArray<Uint8> = [0, ...transaction];
```

Comment:
Does this compile?

Simpler:

`const paddedTx = new Uin8Array([0, ...transaction]);`

Independent of the change: the account for signing is encoded in the transaction? Do we really need to prepend the path length for `cmdSign`?

Reply:
no i gave up after some TS compile struggling figured you would know better as you know the type system that is working pretty well in iov-core but i have not learned it well yet.